File: apps/api/Models/Planejamento.cs
namespace api.Models;
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;

[Table("planejamentos")]
public class Planejamento
{
    [Key]
    [DatabaseGenerated(DatabaseGeneratedOption.Identity)]
    public int ID { get; set; }

    public string Apelido { get; set; }

    public DateOnly DataInicio { get; set; }

    public DateOnly DataFim { get; set; }
    public int IdProfessor { get; set; }

    [ForeignKey("IdProfessor")]
    public Professor Professor { get; set; }

    public string DescicaoPlanejamento { get; set; }

    public string? ObjetivoCurtoPrazo { get; set; }

    public string? ObjetivoMedioPrazo { get; set; }

    public string? ObjetivoLongoPrazo { get; set; }

    public int? ObjetivoCurtoCatalogoId { get; set; }

    [ForeignKey(nameof(ObjetivoCurtoCatalogoId))]
    public PaeeObjetivoCatalogo? ObjetivoCurtoCatalogo { get; set; }

    public int? ObjetivoMedioCatalogoId { get; set; }

    [ForeignKey(nameof(ObjetivoMedioCatalogoId))]
    public PaeeObjetivoCatalogo? ObjetivoMedioCatalogo { get; set; }

    public int? ObjetivoLongoCatalogoId { get; set; }

    [ForeignKey(nameof(ObjetivoLongoCatalogoId))]
    public PaeeObjetivoCatalogo? ObjetivoLongoCatalogo { get; set; }

    public bool DocumentoDeclaradoAssinado { get; set; }

    public string? AssinaturaNomeResponsavel { get; set; }

    public string? AssinaturaCargo { get; set; }

    public ICollection<HabilidadesXPlanejamento> HabilidadesXPlanejamentos { get; set; }

    public ICollection<EstrategiasXPlanejamento> EstrategiasXPlanejamentos { get; set; }

    public ICollection<AvaliacaoXPlanejamento> AvaliacaoXPlanejamentos { get; set; }
    public ICollection<AlunosXPlanejamento> AlunosXPlanejamentos { get; set; }

    public ICollection<PlanejamentoEncontro> Encontros { get; set; } = [];

}

File: apps/web-app/src/types/planejamento.ts
import type { Aluno } from './aluno'
import type { Avaliacao } from './avaliacao'
import type { Estrategia } from './estrategia'
import type { Habilidade } from './habilidade'

/** Linha persistida pela API (.NET DateOnly vira yyyy-MM-dd). */
export interface PaeeEncontro {
  id: number
  dataEnc: string
  textoPlanejado?: string | null
  textoRealizado?: string | null
  habilidadeId?: number | null
  estrategiaId?: number | null
}

export interface PaeeEncontroEntrada {
  dataEnc: string
  textoPlanejado?: string | null
  textoRealizado?: string | null
  habilidadeId?: number | null
  estrategiaId?: number | null
}

export interface PaeeSugestaoDatas {
  datas: string[]
}

export interface PaeeObjetivoCatalogo {
  id: number
  codigo: string
  rotulo: string
  textoModelo: string
  prazo: 'Curto' | 'Medio' | 'Longo' | string
  ordemExibicao: number
}

export interface Planejamento {
  id: number
  apelido: string
  dataInicio: string
  dataFim: string
  descicaoPlanejamento: string
  objetivoCurtoPrazo?: string | null
  objetivoMedioPrazo?: string | null
  objetivoLongoPrazo?: string | null
  objetivoCurtoCatalogoId?: number | null
  objetivoMedioCatalogoId?: number | null
  objetivoLongoCatalogoId?: number | null
  documentoDeclaradoAssinado?: boolean
  assinaturaNomeResponsavel?: string | null
  assinaturaCargo?: string | null
  encontros?: PaeeEncontro[]
  habilidades?: Habilidade[]
  estrategias?: Estrategia[]
  avaliacao?: Avaliacao[]
  alunos?: Aluno[]
}

export interface PlanejamentoAluno {
  id: number
  apelido: string
  dataInicio: string
  dataFim: string
  descicaoPlanejamento: string
  habilidades: Habilidade[]
  estrategias: Estrategia[]
  avaliacao: Avaliacao[]
}

export interface PlanejamentoVinculaAluno {
  idPlanejamento: number
  idAluno: number
}

export interface PlanejamentoVinculaHabilidade {
  idPlanejamento: number
  idHabilidade: number
}

export interface PlanejamentoVinculaEstrategia {
  idPlanejamento: number
  idEstrategia: number
}

export interface PlanejamentoVinculaAvaliacao {
  idPlanejamento: number
  idAvaliacao: number
}

export interface PlanejamentoResponse {
  sucesso: boolean
  mensagens: string[]
  objeto: Planejamento | null
  listaObjetos: Planejamento[]
}
